flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java

```diff
@@ -0,0 +1,224 @@
+	 * @param jedisClusterConfig The configuration of {@link JedisClusterConfig}
+	 * @param redisSinkMapper This used for generate redis command and key value from incoming elements
+     */
+	public RedisSink(JedisClusterConfig jedisClusterConfig, RedisMapper<IN> redisSinkMapper){
+		Preconditions.checkNotNull(jedisClusterConfig, "Redis cluster config should not be Null");
+		Preconditions.checkNotNull(redisSinkMapper, "Redis Mapper can not be null");
+
+		this.redisSinkMapper = redisSinkMapper;
+		this.jedisClusterConfig = jedisClusterConfig;
+		RedisDataTypeDescription dataTypeDescription = redisSinkMapper.getDataTypeDescription();
+		this.redisDataType = dataTypeDescription.getDataType();
+		this.additionalKey = dataTypeDescription.getAdditionalKey();
+
+	}
+
+	/**
+	 * Called when new data arrives to the sink, and forwards it to Redis channel.
+	 *
+	 * @param input The incoming data
+	 */
+	@Override
+	public void invoke(IN input) throws Exception {
+		String key = redisSinkMapper.getKeyFromData(input);
+		String value = redisSinkMapper.getValueFromData(input);
+
+		switch (redisDataType) {
+			case HASH:
+				this.redisCommandsContainer.hset(this.additionalKey, key, value);
+				break;
+
+			case LIST:
+				this.redisCommandsContainer.rpush(key, value);
+				break;
+
+			case SET:
+				this.redisCommandsContainer.sadd(key, value);
+				break;
+
+			case PUBSUB:
+				this.redisCommandsContainer.publish(key, value);
+				break;
+
+			case STRING:
+				this.redisCommandsContainer.set(key, value);
+				break;
+
+			case HYPER_LOG_LOG:
+				this.redisCommandsContainer.pfadd(key, value);
+				break;
+
+			case SORTED_SET:
+				this.redisCommandsContainer.zadd(this.additionalKey, value, key);
+				break;
+
+			default:
+				throw new IllegalArgumentException("Cannot process such data type: " + redisDataType);
+		}
+
+	}
+
+	/**
+	 * Initializes the connection to Redis by either cluster or sentinels or single server
+	 * @throws Exception
+     */
+	@Override
+	public void open(Configuration parameters) throws Exception {
+		if (jedisPoolConfig != null) {
+			this.redisCommandsContainer = RedisCommandsContainerBuilder.build(jedisPoolConfig);
+		} else if (jedisClusterConfig != null) {
+			this.redisCommandsContainer = RedisCommandsContainerBuilder.build(jedisClusterConfig);
+		} else if (jedisSentinelConfig != null){
+			this.redisCommandsContainer = RedisCommandsContainerBuilder.build(jedisSentinelConfig);
+		} else {
+			throw new IllegalArgumentException("Jedis configuration not found");
+		}
+	}
+
+	/**
+	 * Closes commands container
+	 * @throws Exception
+	 */
+	@Override
+	public void close() throws Exception {
+		if (redisCommandsContainer != null){
+			try {
+				redisCommandsContainer.close();
+			} catch (JedisException e) {
+				if (LOG.isErrorEnabled()) {
+					LOG.error("failed to close Redis Commands Container");
+				}
+				throw new RuntimeException("Error while closing Commands Container with error message {}", e);
```

Comment:
Why do you thrown a new `RuntimeException` and not the caught exception `throw e`?